pkg/detectors/aws/aws_test.go

```diff
@@ -227,3 +249,8 @@ func BenchmarkFromData(benchmark *testing.B) {
 		})
 	}
 }
+
+func timeoutContext(timeout time.Duration) context.Context {
+	c, _ := context.WithTimeout(context.Background(), timeout)
```

Comment:
Generally speaking, you always want to make sure a context gets cancelled to cleanup any resources. Since this is for a test I don't think we need to worry about leaks though.